Libraries/DirectMLX.h

```diff
@@ -23,12 +23,16 @@
 #include <type_traits>
 #include <functional>
 
-#if !DMLX_USE_ABSEIL
-    #include <optional>
-#else
+#if DMLX_USE_ABSEIL
     #if __cpp_lib_span
         #include <span>
     #endif
+#elif ((__cplusplus >= 201703L) || (defined(_MSVC_LANG) && (_MSVC_LANG >= 201703L))) && __has_include(<optional>)
```

Comment:
Isn't `__cplusplus` defined in MSVC? Also, `__has_include` is C++17, I don't think this code compiles in an older compiler

Reply:
I was able to compile with __has_include in c++14 successfully.

Reply:
I think `__has_include` is C++17, the fact that some compilers accept it in older standards isn't relevant (it will break in other compilers). If there's a reference that says it's C++14 then it's fine, but I didn't find any.

Also, `<optional>` is always available in C++17, did you add the check to workaround something? It doesn't seem needed, `__cplusplus >= 201703L` should be equivalent to `((__cplusplus >= 201703L) || (defined(_MSVC_LANG) && (_MSVC_LANG >= 201703L))) && __has_include(<optional>)`

Reply:
If you don't know the C++ standard in use and want to use __has_include portably, GCC has a reference on how to do it (https://gcc.gnu.org/onlinedocs/cpp/_005f_005fhas_005finclude.html):

```cpp
#if defined __has_include
#  if __has_include (<stdatomic.h>)
#    include <stdatomic.h>
#  endif
#endif
```